modules/apps/asset/asset-list-web/src/main/resources/META-INF/resources/asset_list/source.jsp

```diff
@@ -234,427 +235,92 @@ List<AssetRendererFactory<?>> classTypesAssetRendererFactories = new ArrayList<>
-			ddmStructureFieldValueInput.value = '';
-
-			<portlet:namespace />toggleSubclasses(true);
-		});
-	}
-
-	var delegate = delegateModule.default;
-
-	delegate(
-		sourcePanel,
-		'click',
-		'.asset-subtypefields-wrapper-enable label',
-		function (event) {
-			var subtypeFieldsFilterEnabledInput = event.delegateTarget.querySelector(
-				'input'
-			);
-
-			var assetSubtypefieldsPopupButtons = document.querySelectorAll(
-				'.asset-subtypefields-popup .btn'
-			);
-
-			if (subtypeFieldsFilterEnabledInput) {
-				Array.prototype.forEach.call(
-					assetSubtypefieldsPopupButtons,
-					function (assetSubtypefieldsPopupButton) {
-						Util.toggleDisabled(
-							assetSubtypefieldsPopupButton,
-							!subtypeFieldsFilterEnabledInput.checked
-						);
-					}
-				);
-			}
-		}
-	);
-
-	Liferay.after('inputmoveboxes:moveItem', function (event) {
-		if (
-			event.fromBox.attr('id') ==
-				'<portlet:namespace />currentClassNameIds' ||
-			event.toBox.attr('id') == '<portlet:namespace />currentClassNameIds'
-		) {
-			<portlet:namespace />toggleSubclasses();
-		}
-	});
-
-	var ddmStructureDisplayFieldValueInput = document.getElementById(
-		'<portlet:namespace />ddmStructureDisplayFieldValue'
-	);
-
-	delegate(sourcePanel, 'click', '.asset-subtypefields-popup', function (event) {
-		var delegateTarget = event.delegateTarget;
-
-		var btn = delegateTarget.querySelector('.btn');
-
-		var url = btn.dataset.href;
-
-		url = Util.addParams(
-			'<portlet:namespace />ddmStructureDisplayFieldValue=' +
-				encodeURIComponent(ddmStructureDisplayFieldValueInput.value),
-			url
-		);
-		url = Util.addParams(
-			'<portlet:namespace />ddmStructureFieldName=' +
-				encodeURIComponent(ddmStructureFieldNameInput.value),
-			url
-		);
-		url = Util.addParams(
-			'<portlet:namespace />ddmStructureFieldValue=' +
-				encodeURIComponent(ddmStructureFieldValueInput.value),
-			url
-		);
-
-		Util.openSelectionModal({
-			customSelectEvent: true,
-			id: '<portlet:namespace />selectDDMStructure' + delegateTarget.id,
-			onSelect: function (selectedItem) {
-				setDDMFields(
-					selectedItem.className,
-					selectedItem.name,
-					selectedItem.value,
-					selectedItem.displayValue,
-					selectedItem.label + ': ' + selectedItem.displayValue
-				);
-			},
-			selectEventName: '<portlet:namespace />selectDDMStructureField',
-			title:
-				'<liferay-ui:message arguments="structure-field" key="select-x" />',
-			url: url,
-		});
-	});
-
-	function setDDMFields(className, name, value, displayValue, message) {
-		ddmStructureFieldNameInput.value = name;
-		ddmStructureFieldValueInput.value = value;
-		ddmStructureDisplayFieldValueInput.value = displayValue;
-
-		var ddmStructureFieldMessageContainer = document.getElementById(
-			'<portlet:namespace />' + className + 'ddmStructureFieldMessage'
-		);
-
-		if (ddmStructureFieldMessageContainer) {
-			ddmStructureFieldMessageContainer.innerHTML = Liferay.Util.escape(
-				message
-			);
-		}
-	}
```

Comment:
This line adds a form input. Please check:

* The input is validated on the processing action

* The backend is protected agains value forging



Refer to documentation: